data_structures/skip_list.cpp

```diff
@@ -0,0 +1,246 @@
+/**
+ * @file skip_list.cpp
+ * @brief Data structure for fast searching and insertion in \f$O(\log n)\f$ time
+ * @details
+ * A skip list is a data structure that is used for storing a sorted list of items with a
+ * help of hierarchy of linked lists that connect increasingly sparse subsequences of the items
+ * 
+ * References used: [GeeksForGeek](https://www.geeksforgeeks.org/skip-list/), [OpenGenus](https://iq.opengenus.org/skip-list) for PseudoCode and Code
+ * @author [enqidu](https://github.com/enqidu)
+ * .
+*/
+
+#include <iostream>
+#include <vector>
+#include <cstring>
+#include <ctime>   
+
+using std::vector;
+using std::endl;
+
+#define MAX_LEVEL 2   ///< Maximum level of skip list
+#define PROBABILITY 0.5      ///< Current probability for "coin toss"
+  
+
+/**
+ *  Node structure [Key][Node*, Node*...]
+*/
+struct Node { 
+    int key;
+    // pointer of value 
+    void* value;
+    // Forward Array points to the neighbour (right)
+    // nodes of the given one in all levels
+    vector<Node*> forward;
```

Comment:
Class variables still not commented correctly. Please refer the following
*  [Doxygen details](https://www.doxygen.nl/manual/docblocks.html)
*  [CONTRIBUTION.md](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#typical-structure-of-a-program)